public/apps/configuration/panels/role-view/role-view.tsx

```diff
@@ -54,12 +78,106 @@ const mappedUserColumns = [
   },
 ];
 
+function createErrorToast(id: string, title: string, text: string): Toast {
+  return {
+    id,
+    color: 'danger',
+    title,
+    text,
+  };
+}
+
+function createUnknownErrorToast(id: string, failedAction: string): Toast {
+  return createErrorToast(
+    id,
+    `Failed to ${failedAction}`,
+    `Failed to ${failedAction}. You may refresh the page to retry or see browser console for more information.`
+  );
+}
+
 export function RoleView(props: RoleViewProps) {
   const duplicateRoleLink = buildHashUrl(ResourceType.roles, Action.duplicate, props.roleName);
 
   const [mappedUsers, setMappedUsers] = useState<MappedUsersListing[]>([]);
   const [errorFlag, setErrorFlag] = useState(false);
   const [selection, setSelection] = useState<MappedUsersListing[]>([]);
+  const [internalUsers, setInternalUsers] = useState<ComboBoxOptions>([]);
+  const [exterIdentities, setExternalIdentities] = useState<ExternalIdentityStateClass[]>([]);
```

Comment:
Is it intended not to be `externalIdentities`?

Reply:
No. It was typo. Corrected.